src/t_stream.c

```diff
@@ -378,37 +387,34 @@ int streamCompareID(streamID *a, streamID *b) {
     return 0;
 }
 
-void streamGetEdgeID(stream *s, int first, streamID *edge_id)
+/* Retrieves the ID of the stream edge entry. An edge is either the first or
+ * the last ID in the stream, and may be a tombstone. To filter out tombstones,
+ * set the'skip_tombstones' argument to 1. */
+void streamGetEdgeID(stream *s, int first, int skip_tombstones, streamID *edge_id)
 {
-    raxIterator ri;
-    raxStart(&ri, s->rax);
-    int empty;
-    if (first) {
-        raxSeek(&ri, "^", NULL, 0);
-        empty = !raxNext(&ri);
-    } else {
-        raxSeek(&ri, "$", NULL, 0);
-        empty = !raxPrev(&ri);
+    streamIterator si;
+    int64_t numfields;
+    int empty = (s->length == 0);
+    streamID min_id, max_id;
+    min_id.ms = 0;
+    min_id.seq = 0;
+    max_id.ms = UINT64_MAX;
+    max_id.seq = UINT64_MAX;
+
+    /* If the stream isn't empty, locate the edge ID (the stream may appear
+     * empty, but may still have tombstones). */
+    if (!empty || !skip_tombstones) {
+        streamIteratorStart(&si,s,&min_id,&max_id,!first);
```

Comment:
This eliminates the need to create `min_id` and `max_id`.
```suggestion
        streamIteratorStart(&si,s,NULL,NULL,!first);
```

Reply:
Agreed, resolved via 5165321a4